tests/python/test_dataset_conversion.py

```diff
@@ -0,0 +1,610 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2022, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import xarray as xr
+import numpy as np
+import pytest
+from smartredis import Dataset,Client
+from smartredis.dataset_utils import DatasetConverter 
+from smartredis.error import *
+
+# ----------Create reference 1D data for 1D tests ---------
+# Generate coordinate data
+longitude_1d = np.linspace(0,360,10)
+# Coordinate tuples 
+longitude_coord_1d = (
+    'x',
+    longitude_1d,
+    { 'x_coord_units':'degrees E', 'x_coord_longname':'Longitude'},
+    None
+) 
+# Data attributes dict 
+data_attributes_1d = {
+    'units':'m/s',
+    'longname':'velocity',
+    'convention':'CF1.5'
+}
+# Generate 1D reference tensor data
+data1d = np.random.random([10])
+# Create 1D reference Xarray for testing 
+ds_1d = xr.DataArray(
+    name="1ddata",
+    data=data1d,
+    dims='x',
+    coords = (longitude_coord_1d,), 
```

Comment:
need to be consistent about spacing around = signs. I prefer a space before and after.